sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -660,6 +662,10 @@ case class ShowPartitionsCommand(
   override def run(sparkSession: SparkSession): Seq[Row] = {
     val catalog = sparkSession.sessionState.catalog
 
+    if (!catalog.tableExists(table)) {
+      throw new AnalysisException(s" Table does not exist")
```

Comment:
Changes done. I used table.unquotedString (similar to what was used in the nearby code. )